### PR TITLE
fix(store): allow `@Select()` decorated properties to be accessed within constructors

### PR DESCRIPTION
**Pull request by arturovt**
Hey, this is the fastest solution I came with. I don't have other ideas. I'm opening a PR so we can discuss the implementation and I'm open to other ideas!

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: packages/store/src/decorators/select/select.ts [hunk 1]
```diff
@@ -23,17 +34,25 @@ export function Select<T>(rawSelector?: T, ...paths: string[]): PropertyDecorato
...
+              mergeMap(() => {
+                const store = localInject(this, Store);
+                const config = localInject(this, NgxsConfig);
+                selector = selector || createSelectorFn(config, name, rawSelector, paths);
+                return createSelectObservable(selector, store);
+              })
+            ));
+          } else {
+            selector = selector || createSelectorFn(null, name, rawSelector, paths);
+            return (this[selectorId] = createSelectObservable(selector, null));
```
Comment (splincode) [inline packages/store/src/decorators/select/select.ts:54]:
```ts
this[selectorId] = createSelectObservable(selector, null);
return this[selectorId];
```

File: packages/store/src/decorators/select/select.ts [hunk 2]
```diff
@@ -23,17 +34,25 @@ export function Select<T>(rawSelector?: T, ...paths: string[]): PropertyDecorato
...
+          }
```
Comment (splincode) [inline packages/store/src/decorators/select/select.ts:55]: what do you think?)

```ts
if (!this[selectorId]) {

  if (ɵivyEnabled) {
    this[selectorId] = injectorNotifier$!.pipe(
      mergeMap(() => {
        const store = localInject(this, Store);
        const config = localInject(this, NgxsConfig);
        selector = selector || createSelectorFn(config, name, rawSelector, paths);
        return createSelectObservable(selector, store);
      })
    ));
  } else {
     selector = selector || createSelectorFn(null, name, rawSelector, paths);
     this[selectorId] = createSelectObservable(selector, null)
  }

}

return this[selectorId];
```

Comment (arturovt) [reply packages/store/src/decorators/select/select.ts:55]: Yeah, I don't mind changing that. That looks better.

I'll wait for Mark's response and I'll then update the code with your suggestion.

**Comment by splincode** [comment]
@markwhitfeld friendly ping))

**Review by markwhitfeld — APPROVED** [review]

**Comment by splincode** [comment]
@markwhitfeld can we release v3.8?

**Comment by markwhitfeld** [comment]
Wait!!! Why did you merge this with failing tests!

**Comment by markwhitfeld** [comment]
Ok, I'll have to fix it on master. There are a number of things to come together for v3.8

**Comment by splincode** [comment]
@markwhitfeld oh, sorry for that, i thought these tests should be ignored

**Pull request merged**